src/index.js

```diff
@@ -41,135 +29,66 @@ if (version.env === "production" && version.branch !== "master") {
     environment: version.env
   }).install();
 }
+export const HEALTHCHECK_INTERVAL_MS = 6000;
 
 const { store, history } = configureStore();
-
-sak(store);
-
-export const HEALTHCHECK_INTERVAL_MS = 6000;
 const PENDING_INITIAL_NETWORK_CHECK = "PENDING_INITIAL_NETWORK_CHECK";
 
-const healthCheck = (dispatch, getState, isInitialHealthcheck = false) => {
-  if (network.isNetworkCheckPending(getState()) === true) {
-    return;
-  }
+let isCheckingConnectivityPromise = null;
+let networkCheckIntervalId = null;
+let checkIfInitiallyLockedIntervalId = null;
 
-  dispatch(setLastNetworkCheckStartAt());
+sak(store);
 
-  if (isInitialHealthcheck) {
-    dispatch(platformReducer.actions.setGlobalFormLockEnabled());
-    dispatch(networkReducer.actions.connecting());
-  }
+const bootstrap = async () => {
+  const { dispatch, getState } = store;
+  period.init(getState);
+  conversion.init(getState);
 
-  Promise.all([Network.checkConnectivity()])
-    .then(async providerType => {
-      const connectedNetworkId = await dispatch(
-        networkReducer.actions.getConnectedNetworkId()
+  await dispatch(
+    platformReducer.actions.web3Initialized(
+      web3.init()
+    )
+  );
+  const onSuccessfulCheck = async ({ dispatch, getState }) => {
+    const accountsList = await promisify(web3p.eth.getAccounts).call();
+    dispatch(
+      networkReducer.actions.setNoProviderConnected(false)
+    );
+    if (accountsList && accountsList.length) {
+      networkCheckIntervalId = PENDING_INITIAL_NETWORK_CHECK;
+      clearInterval(checkIfInitiallyLockedIntervalId);
+      await healthCheck(dispatch, getState, true);
+      networkCheckIntervalId = setInterval(
+        await healthCheck.bind(null, dispatch, getState),
+        HEALTHCHECK_INTERVAL_MS
       );
+    } else {
+      dispatch(platformReducer.actions.accountLocked());
```

Comment:
Actions should be verbs right? I mean that's why they are called actions. 
So  i guess `markAccountAsLocked` or `lockAccount` are better names. Thoughts?